src/main/frontend/fs/capacitor_fs.cljs

```diff
@@ -159,6 +158,19 @@
                       (error-handler error)
                       (log/error :write-file-failed error)))))))))
 
+(defn- get-file-path [dir path]
+  (let [[dir path'] (map #(-> (when %
```

Comment:
The return value of the when block could be nil which would get passed to js/decodeURI. I don't think we want that as that would result in "null". If you want to avoid calling js/decodeURI and pass nil to dir and path: `(some-> % (string/replace ...) js/decodeURI)`

Reply:
@logseq-cldwalker Thanks a lot for your review. `some->` is much better here. 
Parameters here are:
- Normally, `dir` is never a nil (always be e.g. `file:///private/var/mobile/Library/Mobile%20Documents/iCloud~com~logseq~logseq/Documents/`, encoded URL)
- `path`
     -  a full path (e.g. `file:///private/var/mobile/Library/Mobile Documents/iCloud~com~logseq~logseq/Documents/pages/pages-metadata.edn`, decoded URL)
     - or a relative path (e.g. `/journal/2022_01_28.md`)
     - or `nil` (only when adding a new refresh graph). 

As you can see, replacing `///` with `/` is for later path match, and even though `path'` gets `null`, there will never be any issue here.

Will use `some->` here.

BTW, we can take care of the `///` thing in other place in the future.

Reply:
Thanks for the explanation of the parameters. I'll write some tests for these on Monday with them. For the case of the nil path, what should the fn return?

> If path is nil, removing when causes(string/replace path ''///'' ''/'') to run an error, so I think we can change it like this.

`some->` guards against nil errors by not running sexpr's to the right if the current value is nil. You can confirm this with babashka - `bb '(prn (some-> nil #(str/replace % "///" "/")))'`

Reply:
Thanks. When path nils, the function returns the encoded graph dir, e.g.  `file:/private/var/mobile/Library/Mobile%20 Documents/iCloud~com~logseq~logseq/Documents/`.

Nice to know the tip of `some->` for error guards. I've pushed a commit to address this.